lib/web3/syncing.js

```diff
@@ -63,23 +59,18 @@ var pollSyncing = function(self) {
         });
     };
```

Comment:
Thanks for looking over it.
The isSyncing function is used in production and works well and as intended. There is probably improvement potential when using multiple instances.

Thought it looks like you don't fully understand its functionality.

There is also a `web3.eth.syncing` property, which simply returns you the value of the `eth_syncing`, though the `isSyncing` is meant as a convenient function for dapps to detect sync starts and stops and allow them to display appropriate information about the progress.

The callback is supposed to only call `true` once when a sync starts, so the dapp can stop its operations and then repeatly on every sync progress. It does matter here if it returns duplicates here, as mostly it won't (sync progress faster than 500ms)
And then it should call `false` once, when its finished. so the app can regain app operation.

The necessity of this function is to prevent dapps form being swamped with new blocks, and then send back a bunch of requests.

You're right, there might be issues when you instantiate multiple isSyncing, thats why i add the `addCallback`, which is what you should use (Though normally you will only use one instance anyway).
But i can make it so that there is only one instance of it and it uses the add callback.

> prototype constructor returns this. It's redundant

I see

> setTimeout(function() {}, 0) 

Makes sense, i was not sure..

> remove lastSyncState. Always return what api returns.
> remove callback(null, true). Imo it's not necessary. Callback function must check typeof result anyway

I don't agree with any of the necessary to tell the user when it starts, see this example:
```js
web3.eth.isSyncing(function(error, sync){
    if(!error) {
        // stop all app activity
        if(sync === true) {
           // we use `true`, so it stops all filters, but not the web3.eth.syncing polling
           web3.reset(true);

        // show sync info
        } else if(sync) {
           console.log(sync.currentBlock);

        // re-gain app operation
        } else {
            // run your app init function...
        }
    }
});
```

> fix IsSyncing tests and/or revert the changes in IsSyncing prototype.
Should be working, when you fixed the uncleared test, right?